src/sampling/regular.jl

```diff
@@ -172,4 +172,28 @@ end
 function sample(rng::AbstractRNG, grid::CartesianGrid,
                 method::RegularSampling)
   sample(rng, boundingbox(grid), method)
-end
\ No newline at end of file
+end
+
+function sample(::AbstractRNG, torus::Torus{T},
+                method::RegularSampling) where {T}
+  sz = fitdims(method.sizes, paramdim(torus))
+  R, r = radii(torus)
+  kxy = R^2 - r^2
+  kz = √kxy * r
+  s = √kxy / r 
+
+  V = T <: AbstractFloat ? T : Float64
+  umin, umax = V(-s * π), V(s * π)
+  vmin, vmax = V(-π), V(π)
+  urange = range(umin, stop=umax, length=sz[1]+1)[1:sz[1]]
+  vrange = range(vmin, stop=vmax, length=sz[2]+1)[1:sz[2]]
```

Comment:
Please try to follow the convention in other regular samplers. This indexing with [1:sz[1]] looks strange

Reply:
But when this is periodic on a range e.g. `0 - 2pi` one has to drop one of the numbers at the edge of the range otherwise there's a duplicated vertex in the mesh.

Reply:
Yes, but we do that in the range itself. See Cylinder for example where the correct range is constructed for one of the dimensions.